cmd/lifecycle-main.go

```diff
@@ -0,0 +1,134 @@
+	Prefix  string      `xml:"Prefix,omitempty" json:"Prefix,omitempty"`
+	Tags    []tagFilter `xml:"Tag,omitempty" json:"Tags,omitempty"`
+}
+
+type lifecycleRuleFilter struct {
+	XMLName xml.Name              `xml:"Filter" json:"-"`
+	And     *lifecycleAndOperator `xml:"And,omitempty" json:"And,omitempty"`
+	Prefix  string                `xml:"Prefix,omitempty" json:"Prefix,omitempty"`
+	Tag     *tagFilter            `xml:"Tag,omitempty" json:"-"`
+}
+
+type lifecycleExpiration struct {
+	XMLName          xml.Name   `xml:"Expiration,omitempty" json:"-"`
+	ExpirationDate   *time.Time `xml:"Date,omitempty" json:"Date,omitempty"`
+	ExpirationInDays int        `xml:"Days,omitempty" json:"Days,omitempty"`
+}
+
+type lifecycleRule struct {
+	XMLName                        xml.Name                        `xml:"Rule,omitempty" json:"-"`
+	AbortIncompleteMultipartUpload *abortIncompleteMultipartUpload `xml:"AbortIncompleteMultipartUpload,omitempty" json:"AbortIncompleteMultipartUpload,omitempty"`
+	Expiration                     *lifecycleExpiration            `xml:"Expiration,omitempty" json:"Expiration,omitempty"`
+	ID                             string                          `xml:"ID" json:"ID"`
+	RuleFilter                     *lifecycleRuleFilter            `xml:"Filter,omitempty" json:"Filter,omitempty"`
+	NoncurrentVersionTransition    *noncurrentVersionTransition    `xml:"NoncurrentVersionTransition,omitempty" json:"NoncurrentVersionTransition,omitempty"`
+	NoncurrentVersionTransitions   []noncurrentVersionTransition   `xml:"NoncurrentVersionTransitions,omitempty" json:"NoncurrentVersionTransitions,omitempty"`
+	Prefix                         string                          `xml:"Prefix,omitempty" json:"Prefix,omitempty"`
+	Status                         string                          `xml:"Status" json:"Status"`
+	Transition                     *lifecycleTransition            `xml:"Transition,omitempty" json:"Transition,omitempty"`
+	Transitions                    []lifecycleTransition           `xml:"Transitions,omitempty" json:"Transitions,omitempty"`
+	TagFilters                     []tagFilter                     `xml:"Tag,omitempty" json:"Tags,omitempty"`
+}
+
+type lifecycleConfiguration struct {
+	XMLName xml.Name        `xml:"LifecycleConfiguration,omitempty" json:"-"`
+	Rules   []lifecycleRule `xml:"Rule"`
+}
+
+// checkIlmMainSyntax - validate arguments passed by a user
+func checkIlmMainSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) == 0 || len(ctx.Args()) > 1 {
+		cli.ShowCommandHelp(ctx, "")
+		os.Exit(globalErrorExitStatus)
+	}
+}
+
+func mainLifecycle(ctx *cli.Context) error {
+	checkIlmMainSyntax(ctx)
+	setColorScheme()
+	args := ctx.Args()
+	objectURL := args.Get(0)
+	var pErr *probe.Error
+	clnt, pErr := newClient(objectURL)
+	if pErr != nil {
+		cli.ShowCommandHelp(ctx, "")
+		return pErr.Trace(objectURL).ToGoError()
+	}
+
+	_, pErr = clnt.GetBucketLifecycle()
+	if pErr != nil {
+		cli.ShowCommandHelp(ctx, "")
```

Comment:
This should only be printed if input syntax validation is wrong not for all API errors @BigUstad